src/mngt/ocf_mngt_common.c

```diff
@@ -54,41 +54,46 @@ void cache_mngt_core_remove_from_cleaning_pol(ocf_core_t core)
 /* Deinitialize core metadata in attached metadata */
 void cache_mngt_core_deinit_attached_meta(ocf_core_t core)
 {
-	int retry = 1;
-	uint64_t core_size = 0;
-	ocf_cleaning_t clean_pol_type;
 	ocf_cache_t cache = ocf_core_get_cache(core);
 	ocf_core_id_t core_id = ocf_core_get_id(core);
+	ocf_core_id_t iter_core_id;
+	ocf_cache_line_t line, hash;
+	uint32_t num_hash = cache->device->hash_table_entries;
 
-	core_size = ocf_volume_get_length(&core->volume);
-	if (!core_size)
-		core_size = ~0ULL;
+	for (hash = 0; hash < num_hash; hash++) {
+		ocf_metadata_lock_hash_wr(&cache->metadata.lock, hash);
 
-	ocf_metadata_start_exclusive_access(&cache->metadata.lock);
+		line = ocf_metadata_get_hash(cache, hash);
+		while (line != cache->device->collision_table_entries) {
+			ocf_metadata_get_core_info(cache, line, &iter_core_id,
+					NULL);
 
-	clean_pol_type = cache->conf_meta->cleaning_policy_type;
-	while (retry) {
-		retry = 0;
-		if (cleaning_policy_ops[clean_pol_type].purge_range) {
-			retry = cleaning_policy_ops[clean_pol_type].purge_range(cache,
-					core_id, 0, core_size);
-		}
+			if (iter_core_id == core_id) {
```

Comment:
suggestion to reduce number of indentation levels:
if (iter_core_id != core_id) {
   line = ocf_metadata_get_collision_next(cache, line);
   continue;
}